Source/WebCore/page/ContextMenuController.cpp

```diff
@@ -998,29 +1006,39 @@ void ContextMenuController::populate()
                 appendItem(StopItem, m_contextMenu.get());
                 appendItem(ReloadItem, m_contextMenu.get());
 #else
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(-1))
-                    appendItem(BackItem, m_contextMenu.get());
-
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(1))
-                    appendItem(ForwardItem, m_contextMenu.get());
-
-                // use isLoadingInAPISense rather than isLoading because Stop/Reload are
-                // intended to match WebKit's API, not WebCore's internal notion of loading status
-                if (loader.documentLoader()->isLoadingInAPISense())
-                    appendItem(StopItem, m_contextMenu.get());
-                else
-                    appendItem(ReloadItem, m_contextMenu.get());
+                if (frame->isMainFrame() || (frame->ownerElement() && !frame->ownerElement()->document().isPDFDocument())) {
+                    if (frame->page() && frame->page()->backForward().canGoBackOrForward(-1))
+                        appendItem(BackItem, m_contextMenu.get());
+
+                    if (frame->page() && frame->page()->backForward().canGoBackOrForward(1))
+                        appendItem(ForwardItem, m_contextMenu.get());
+                    
+                    // use isLoadingInAPISense rather than isLoading because Stop/Reload are
+                    // intended to match WebKit's API, not WebCore's internal notion of loading status
+                    if (loader.documentLoader()->isLoadingInAPISense())
+                        appendItem(StopItem, m_contextMenu.get());
+                    else
+                        appendItem(ReloadItem, m_contextMenu.get());
+                }
 #endif
                 }
 
-                if (frame->page() && !frame->isMainFrame())
+                if (frame->ownerElement() && !frame->isMainFrame() && !frame->ownerElement()->document().isPDFDocument()) 
                     appendItem(OpenFrameItem, m_contextMenu.get());
-
+                
                 if (!ShareMenuItem.isNull()) {
                     appendItem(*separatorItem(), m_contextMenu.get());
                     appendItem(ShareMenuItem, m_contextMenu.get());
                 }
             }
+#if ENABLE(PDFJS)
+            if (frame->ownerElement() && frame->ownerElement()->document().isPDFDocument()) {
```

Comment:
This can be `if (auto* ownerElement = frame->ownerElement(); ownerElement->document().isPDFDocument())`.

Reply:
Updated if statement